File: backend/SpareHub/Domain/Vessel.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Domain;

public class Vessel
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; init; }
    public int OwnerId { get; init; }
    public required string Name { get; init; }
    public string? ImoNumber { get; init; }
    public string? Flag { get; init; }

    public required Owner Owner { get; init; }

    [JsonIgnore]
    public ICollection<VesselAtPort> VesselAtPorts { get; set; } = new List<VesselAtPort>();
}

File: frontend/src/interfaces/vessel.ts
import { Owner } from './owner';

export interface Vessel {
    id: string;
    owner_id: string;
    name: string;
    imoNumber: string;
    flag: string;
    owner?: Owner;
}

export interface VesselDetail extends Vessel {
    id: string;
    owner_id: string;
    name: string;
    imoNumber: string;
    flag: string;
}

export interface VesselRequest {
    name: string;
    imoNumber: string;
    flag: string;
    ownerId: string;
} 